lib/ui/text.dart

```diff
@@ -431,6 +431,85 @@ enum TextDecorationStyle {
   wavy
 }
 
+/// Defines how the paragraph will handle the ascent of the first line and
+/// descent of the last line. These lines are referred to as "boundary" lines.
+class BoundaryLineHeightBehavior {
+
+  /// Creates a new BoundaryLineHeightBehavior object.
+  ///
+  ///  * first: When true, the [TextStyle.height] modifier will be applied to
+  ///    to the ascent of the first line. When false, the font's default ascent
+  ///    will be used.
+  ///  * last: When true, the [TextStyle.height] modifier will be applied to
+  ///    to the descent of the last line. When false, the font's default descent
+  ///    will be used.
+  ///
+  /// All properties default to true (height modificaitons applied as normal).
+  BoundaryLineHeightBehavior({
+    this.first = true,
+    this.last = true,
+  });
+
+  /// Creates a new BoundaryLineHeightBehavior object from an encoded form.
+  ///
+  /// See [encode] for the creation of the encoded form.
+  BoundaryLineHeightBehavior.fromEncoded(int encoded) : first = (encoded & 0x1) > 0,
+                                                        last = (encoded & 0x2) > 0;
+
+
+  /// Whether to apply the [TextStyle.height] modifier or not to the ascent of
+  /// the first line in the paragraph.
+  ///
+  /// When true, the [TextStyle.height] modifier will be applied to to the ascent
+  /// of the first line. When false, the font's default ascent will be used and
+  /// the [TextStyle.height] will have no effect on the ascent of the first line.
+  ///
+  /// Defaults to true (height modificaitons applied as normal).
+  final bool first;
```

Comment:
"first" is not a yes/no question. It would be more intuitive with a name like "padFirstLine" or something.

Reply:
or "applyAscentToFirstLine" and "applyDescentToLastLine" maybe

Reply:
Changed to `applyHeightToFirstLineAscent` A bit long but very descriptive.